From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andre Braga <code@andrebraga.com>
Date: Sat, 29 Aug 2026 11:27:39 +0200
Subject: arm64: dts: qcom: cq8725s-ayn: wire USB-C SS through NB7VPQ904M
 redriver

> X-Git-Archeology: > recovered message: > Bring over the patches ROCKNIX still carries on top of the set imported
> X-Git-Archeology: > recovered message: > in the bringup: the integrated-jack USBSS routing with the HPH PCM HiFi
> X-Git-Archeology: > recovered message: > path, USB-C redriver and probe-defer fixes, the hv-haptics driver with
> X-Git-Archeology: > recovered message: > the rsinput force feedback bridge, UFS support for the SM8750
> X-Git-Archeology: > recovered message: > controller and the thermal, capacity and RTC DT bits. Patches aimed at
> X-Git-Archeology: > recovered message: > other devices (KONKR, Ayaneo) are left out; the collided 0052 rpmhpd
> X-Git-Archeology: > recovered message: > patch is renumbered to 0053.
> X-Git-Archeology: > recovered message: > Signed-off-by: SuperKali <hello@superkali.me>
> X-Git-Archeology: - Revision d7ac3357efedd9f44916dc489e9727e95a34f0b9: https://github.com/armbian/build/commit/d7ac3357efedd9f44916dc489e9727e95a34f0b9
> X-Git-Archeology:   Date: Sat, 29 Aug 2026 11:27:39 +0200
> X-Git-Archeology:   From: SuperKali <hello@superkali.me>
> X-Git-Archeology:   Subject: kernel/sm8750: sync the remaining ROCKNIX patches
> X-Git-Archeology:
---
 arch/arm64/boot/dts/qcom/cq8725s-ayn-common.dtsi | 35 +++++++++-
 1 file changed, 33 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/qcom/cq8725s-ayn-common.dtsi b/arch/arm64/boot/dts/qcom/cq8725s-ayn-common.dtsi
index 111111111111..222222222222 100644
--- a/arch/arm64/boot/dts/qcom/cq8725s-ayn-common.dtsi
+++ b/arch/arm64/boot/dts/qcom/cq8725s-ayn-common.dtsi
@@ -213,7 +213,7 @@ port@1 {
 					reg = <1>;
 
 					pmic_glink_ss_in: endpoint {
-						remote-endpoint = <&usb_dp_qmpphy_out>;
+						remote-endpoint = <&redriver_ss_out>;
 					};
 				};
 
@@ -888,6 +888,37 @@ spk_amp_r: spk_amp_r@35 {
 		awinic,sync-flag;
 		sound-name-prefix = "SPK_R";
 	};
+
+	typec_mux_redriver: typec-mux@1c {
+		compatible = "onnn,nb7vpq904m";
+		reg = <0x1c>;
+
+		vcc-supply = <&vreg_l15b_1p8>;
+
+		retimer-switch;
+		orientation-switch;
+
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			port@0 {
+				reg = <0>;
+
+				redriver_ss_out: endpoint {
+					remote-endpoint = <&pmic_glink_ss_in>;
+				};
+			};
+
+			port@1 {
+				reg = <1>;
+
+				redriver_ss_in: endpoint {
+					remote-endpoint = <&usb_dp_qmpphy_out>;
+				};
+			};
+		};
+	};
 };
 
 &iris {
@@ -1220,7 +1251,7 @@ &usb_dp_qmpphy {
 };
 
 &usb_dp_qmpphy_out {
-	remote-endpoint = <&pmic_glink_ss_in>;
+	remote-endpoint = <&redriver_ss_in>;
 };
 
 &usb_dwc3_hs {
-- 
Armbian

